### PR TITLE
P4 1330 oca assign person to allocation tests

### PR DESCRIPTION
**Pull request by solidgoldpig**
Adds tests for code added in #539 

### Checklist

- [x] If adding new environment variables, they have been documented in the [README](ministryofjustice/hmpps-book-secure-move-frontend/blob/master/README.md)
- [x] No new [Personally Identifiable Information (PII)](https://support.google.com/analytics/answer/6366371?hl=en) is being sent via analytics

### CODE REVIEW

File: app/allocation/controllers/assign/save.test.js [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+      )
+    })
+
+    describe('When mixing in create controller', function() {
+      it('should copy process from CreatePersonSearch', function() {
+        expect(controller.process).to.exist.and.equal(MixinProto.process)
+      })
+
+      it('should only have the expected methods of its own', function() {
+        const ownMethods = ['saveValues', 'errorHandler', 'successHandler']
```
Comment (teneightfive) [inline app/allocation/controllers/assign/save.test.js:26]: There only seem to be tests for `saveValues()`, can you add tests for the `errorHandler` and `successHandler` while you're here please

Comment (solidgoldpig) [reply app/allocation/controllers/assign/save.test.js:26]: Added

File: app/allocation/controllers/assign/base.test.js [hunk 2]
```diff
@@ -0,0 +1,232 @@
...
+      expect(nextSpy).to.be.calledOnceWithExactly()
+    })
+  })
+
+  describe('#setMove()', function() {
+    let req
+    let nextSpy
+    let res = {}
+
+    // const person = { id: '__person__' }
```
Comment (teneightfive) [inline app/allocation/controllers/assign/base.test.js:119]: Rogue comment

File: app/allocation/fields/special_vehicle_check.js [hunk 3]
```diff
@@ -0,0 +1,23 @@
+const specialVehicleCheck = {
```
Comment (teneightfive) [inline app/allocation/fields/special_vehicle_check.js:1]: Could we use kebabcase in the filename please. Keeps it consistent with the rest of the codebase

File: app/allocation/fields/index.js [hunk 4]
```diff
@@ -24,28 +25,9 @@ const createFields = {
...
-      value: 'true',
-      text: 'fields::special_vehicle_check.items.yes.label',
-    },
-    {
-      value: 'false',
-      text: 'fields::special_vehicle_check.items.no.label',
-    },
-  ],
+const assignFields = {
+  ...moveCreateFields,
```
Comment (teneightfive) [inline app/allocation/fields/index.js:29]: Should we maybe be deepcloning this?

File: app/allocation/controllers/assign/base.test.js (hunk 2 above)
Comment (solidgoldpig) [reply app/allocation/controllers/assign/base.test.js]: already zapped!

File: app/allocation/fields/special_vehicle_check.js (hunk 3 above)
Comment (solidgoldpig) [reply app/allocation/fields/special_vehicle_check.js]: 👍

File: app/allocation/fields/index.js (hunk 4 above)
Comment (solidgoldpig) [reply app/allocation/fields/index.js]: Sorted

**Pull request merged**